File: pythonScript/wiki/python/MongoDBConnector.py
import pymongo
import time
import asyncio

class MongoDBConnector:
    def __init__(self, collection):
        try: 
            self.client = pymongo.MongoClient("mongodb+srv://S09P31B205:z5HxUpl4gB@ssafy.ngivl.mongodb.net/S09B31B205?authSource=admin")
            time.sleep(60)
            # print(f"connect check : {self.client}")
            self.db = self.client["S09P31B205"]
            self.collection = self.db[collection]
        except pymongo.errors.ConnectionFailure as e:
            print(f"Failed to connect to MongoDB: {e}")

    def insert_data(self, data):
        # 데이터를 MongoDB에 삽입
        result = self.collection.insert_one(data)
        return result.inserted_id
    
    def find_data(self, data):
        #데이터 검색
        result = self.collection.find_one({"title":data})         
        return result                                                                                                                                                                            
    
    def find_all(self):
        result = self.collection.find({})
        time.sleep(60)
        return result

File: pythonScript/wiki/node/MongoDBConnector.js
const { MongoClient } = require('mongodb');
const delay = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

class MongoDBConnector {
  constructor(collection) {
    this.collectionName = collection;
    this.client = null;
    this.db = null;
    this.collection = null;
  }

  async connect() {
    try {
      this.client = new MongoClient('mongodb+srv://S09P31B205:z5HxUpl4gB@ssafy.ngivl.mongodb.net/S09B31B205', {
        useNewUrlParser: true,
        useUnifiedTopology: true,
      });
      await this.client.connect();
      this.db = this.client.db('S09P31B205');
      this.collection = this.db.collection(this.collectionName);
    } catch (error) {
      console.error('Failed to connect to MongoDB:', error);
    }
  }

  async insertData(data) {
    if (!this.collection) {
      await this.connect();
    }

    try {
      const result = await this.collection.insertOne(data);
      return result.insertedId;
    } catch (error) {
      console.error('Failed to insert data:', error);
      return null;
    }
  }

  async findData(data) {
    if (!this.collection) {
      await this.connect();
    }

    try {
      const result = await this.collection.findOne({ title: data });
      return result;
    } catch (error) {
      console.error('Failed to find data:', error);
      return null;
    }
  }

  async findAll() {
    if (!this.collection) {
      await this.connect();
    }

    try {
      const result = await this.collection.find({}).toArray();
      return result;
    } catch (error) {
      console.error('Failed to find all data:', error);
      return [];
    }
  }
}

// Example usage:
const connector = new MongoDBConnector('summary_overview');
(async () => {
  await connector.connect();
  const allData = await connector.findAll();
  console.log('All data:', allData);
})();
